## 1. 목표)
### 대학교 합격률을 예측해보자.

In [1]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

df = pd.read_csv('C:/Users/katenam/DataDrivenReport/Admission_Predict.csv')
df.head()

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0           1        337          118                  4  4.5   4.5  9.65   
1           2        324          107                  4  4.0   4.5  8.87   
2           3        316          104                  3  3.0   3.5  8.00   
3           4        322          110                  3  3.5   2.5  8.67   
4           5        314          103                  2  2.0   3.0  8.21   

   Research  Chance of Admit   
0         1              0.92  
1         1              0.76  
2         1              0.72  
3         1              0.80  
4         0              0.65

In [4]:
df = df.rename(columns={'Serial No.': 'SerialNo', 'GRE Score': 'GREScore',
                        'TOEFL Score': 'TOEFLScore', 'University Rating': 'UniversityRating',
                        'SOP': 'SOP', 'LOR ': 'LOR', 'CGPA': 'CGPA', 'Research': 'Research',
                        'Chance of Admit ': 'ChanceOfAdmit'})
df.columns

Index(['SerialNo', 'GREScore', 'TOEFLScore', 'UniversityRating', 'SOP', 'LOR',
       'CGPA', 'Research', 'ChanceOfAdmit'],
      dtype='object')

In [5]:
x_data = df[['GREScore', 'TOEFLScore', 'UniversityRating', 'SOP', 'LOR', 'CGPA','Research']]
target = df["ChanceOfAdmit"]

In [2]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

In [6]:
# Split data for test and train the model.
x_train, x_test, y_train, y_test = train_test_split(x_data,target,random_state=0,test_size=0.2)

In [7]:
#object
linreg = LinearRegression()
# fiting our data for training
linreg.fit(x_train,y_train)

LinearRegression()

In [8]:
# our model is ready to predict y.
y_pred = linreg.predict(x_test)
y_pred  

array([0.69791327, 0.69343926, 0.77882728, 0.61577391, 0.72281999,
       0.60286273, 0.7067783 , 0.64104331, 0.87037545, 0.91874738,
       0.53830841, 0.88537227, 0.70839345, 0.45744365, 0.85464724,
       0.62329657, 0.63034415, 0.81411912, 0.58479826, 0.73198733,
       0.8755188 , 0.8436008 , 0.65964574, 0.45723204, 0.78390781,
       0.58302231, 0.49932541, 0.64153708, 0.88778116, 0.64950902,
       0.64065401, 0.74057509, 0.74268991, 0.55702869, 0.78303581,
       0.77159575, 0.65140153, 0.86244822, 0.63038997, 0.92925287,
       0.72728695, 0.67164181, 0.70967833, 0.80792068, 0.8242496 ,
       0.6540426 , 0.56311848, 0.69211512, 0.60750179, 0.60172386,
       0.67514274, 0.80138513, 0.65205933, 0.87968402, 0.72302425,
       0.75457701, 0.70602419, 0.73465988, 0.75784394, 0.82956036,
       0.77359675, 0.47343091, 0.63752277, 0.52286953, 0.84258874,
       0.82142312, 0.73664582, 0.86699812, 0.76933952, 0.75453984,
       0.59900479, 0.82915301, 0.82822468, 0.60736795, 0.93236

- #### y_pred라는 결과값을 보면 첫 번째 값 0.6979는 x_test의 각 변수들(GRE, SOP, SPA 등)의 조합이 69.79%라는 y값을 예측했다는 것이다.<br/><br/> 즉, y_test 개수만큼의 값을 예측하고 예측한 값과 y_test 실제값을 비교하여 정확도를 측정하게 된다.


In [9]:
from sklearn.metrics import r2_score

print('R-squared score:', r2_score(y_test, y_pred))

R-squared score: 0.7355078738145215


- #### <span style="color:blueviolet">R-squared(결정계수)</span>는 실제값(y)과 예측값(y_pred) 차이를 제곱하여 평균한 값의 비율을 나타내는 지표이다. 

### 2) 예측
- #### 직접 각 변수별 점수를 넣고 합격에 대한 확률을 알아보자

In [13]:
print('The chance of you getting an admit in the US is {}%'.format(round(linreg.predict([[305, 108, 4, 4.5, 4.5, 8.35, 0]])[0]*100, 1)))

The chance of you getting an admit in the US is 67.9%


In [14]:
print('The chance of you getting an admit in the US is {}%'.format(round(linreg.predict([[370, 108, 4, 4.5, 4.5, 8.35, 1]])[0]*100, 1)))

The chance of you getting an admit in the US is 83.8%
